core/domain/exp_domain.py

```diff
@@ -3138,6 +3180,54 @@ def _convert_states_v51_dict_to_v52_dict(
 
         return states_dict
 
+    @classmethod
+    def _convert_states_v52_dict_to_v53_dict(
+        cls, states_dict: Dict[str, state_domain.StateDict]
+    ) -> Dict[str, state_domain.StateDict]:
+        """Converts from version 52 to 53. Version 53 makes sure the links used
+        in the non-interactive link component are all 'https' links.
+
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+
+        for state_dict in states_dict.values():
+            valid_html = (
+                cls.fix_non_interactive_links(state_dict['content']['html']))
+
+            state_dict['content']['html'] = valid_html
+
+        for state_dict in states_dict.values():
+            translations_mapping = (
+                state_dict['written_translations']['translations_mapping'])
+            new_translations_mapping = {}
+            for content_id, translation_item in translations_mapping.items():
+                new_translations_mapping[content_id] = translation_item
+                if content_id != 'content':
+                    continue
+                for language_code in translations_mapping[content_id]:
+                    written_translation = translation_item[language_code]
+                    # Here, we are narrowing down the type from
+                    # Union[List[str], str] to str.
+                    assert isinstance(
+                        written_translation['translation'], str
+                    )
+                    correct_html = cls.fix_non_interactive_links(
+                        written_translation['translation'])
+
+                    written_translation['translation'] = correct_html
+                    translation_item[language_code] = written_translation
+                            
+            state_dict['written_translations']['translations_mapping'] = (
+                new_translations_mapping)
+
+        return states_dict
```

Comment:
@Lawful2002 Are we expecting to only fix link issues in state.content and translations? (And not the hints, feedback, solutions etc.)

Reply:
We should actually do it in all the places, not just content/translations.

Reply:
@Lawful2002 You can use the `State.convert_html_fields_in_state` function instead of rewriting the traversal function.

https://github.com/oppia/oppia/blob/8eeef29fc5eba28e22b677fcb6951bbfe20e1eee/core/domain/state_domain.py#L5128-L5138